### PR TITLE
Update python-telegram-bot to 13

### PR DESCRIPTION
**Pull request by tehcneko**

### CODE REVIEW

**Comment by DO97** [comment]
Great job!

``actions.py is missing``

File: bot.py [hunk 1]
```diff
@@ -387,19 +387,19 @@ def start_game(bot, update, args, job_queue):
             def send_first():
```
Comment (DO97) [inline bot.py:387]: Line 386 - ``TelegramDeprecationWarning: The @run_async decorator is deprecated. Use the `run_async` parameter of your Handler or `Dispatcher.run_async` instead``

File: bot.py [hunk 2]
```diff
@@ -303,12 +303,12 @@ def select_game(bot, update):
     def selected(bot):
```
Comment (DO97) [inline bot.py:303]: Line 302 - ``TelegramDeprecationWarning: The @run_async decorator is deprecated. Use the `run_async` parameter of your Handler or `Dispatcher.run_async` instead``

Line 321 - Missing "context" in ``selected(bot)``

**Comment by DO97** [comment]
When trying to play a Sanic mode game:

**Comment by DO97** [comment]
Another Job Queue related error

**Comment by DO97** [comment]
Fix (credits to @jh0ker)
```
    def selected():
        back = [[InlineKeyboardButton(text=_("Back to last group"),
                                      switch_inline_query='')]]
        context.bot.answerCallbackQuery(update.callback_query.id,
                                text=_("Please switch to the group you selected!"),
                                show_alert=False,
                                timeout=TIMEOUT)

        context.bot.editMessageText(chat_id=update.callback_query.message.chat_id,
                            message_id=update.callback_query.message.message_id,
                            text=_("Selected group: {group}\n"
                                   "<b>Make sure that you switch to the correct "
                                   "group!</b>").format(
                                group=gm.userid_current[user_id].game.chat.title),
                            reply_markup=InlineKeyboardMarkup(back),
                            parse_mode=ParseMode.HTML,
                            timeout=TIMEOUT)

    dispatcher.run_async(selected)
```

Fix (credits to @jh0ker)
```
            def send_first():
                """Send the first card and player"""

                context.bot.sendSticker(chat.id,
                                sticker=c.STICKERS[str(game.last_card)],
                                timeout=TIMEOUT)

                context.bot.sendMessage(chat.id,
                                text=first_message,
                                reply_markup=InlineKeyboardMarkup(choice),
                                timeout=TIMEOUT)

            dispatcher.run_async(send_first)
```

**Comment by DO97** [comment]
Fix (credits to @jh0ker)

Line 153 - internationalization.py
```
def _user_chat_from_update(update):
    user = update.effective_user
    chat = update.effective_chat

    if chat is None and user is not None and user.id in gm.userid_current:
        chat = gm.userid_current.get(user.id).game.chat

    return user, chat
```

**Comment by DO97** [comment]
```
2022-03-12 15:52:42,700 - utils - ERROR - Exception while handling an update:
Traceback (most recent call last):
  File "/home/U/UBot/venv/lib/python3.9/site-packages/telegram/ext/dispatcher.py", line 555, in process_update
    handler.handle_update(update, self, check, context)
  File "/home/U/UBot/venv/lib/python3.9/site-packages/telegram/ext/handler.py", line 198, in handle_update
    return self.callback(update, context)
  File "<string>", line 2, in wrapped
  File "/home/U/UBot/venv/lib/python3.9/site-packages/pony/orm/core.py", line 519, in new_func
    result = func(*args, **kwargs)
  File "/home/U/UBot/internationalization.py", line 115, in wrapped
    result = func(update, context, *pargs, **kwargs)
  File "/home/U/UBot/bot.py", line 257, in join_game
    chat = update.message.chat
AttributeError: 'NoneType' object has no attribute 'chat'
```
In order to avoid this kind of error, you have to exclude via Filters ("`from telegram.ext import Filters`") and avoid updates in case of edited commands/messages.

This applies to all CommandHandler(s) and MessageHandler(s) in:
`bot.py`
`settings.py`
`simple_commands.py`

Example:
```
dispatcher.add_handler(CommandHandler('settings', show_settings, filters=~Filters.update.edited_message))
dispatcher.add_handler(MessageHandler(~Filters.update.edited_message
                                      & Filters.regex('^([' + '📊' +
```

**Comment by DO97** [comment]
> Fix (credits to @jh0ker)
> 
> Line 153 - internationalization.py
> 
> ```
> def _user_chat_from_update(update):
>     user = update.effective_user
>     chat = update.effective_chat
> 
>     if chat is None and user is not None and user.id in gm.userid_current:
>         chat = gm.userid_current.get(user.id).game.chat
> 
>     return user, chat
> ```

```Traceback (most recent call last):
  File "/home/U/UBot/venv/lib/python3.9/site-packages/apscheduler/executors/base.py", line 125, in run_job
    retval = job.func(*job.args, **job.kwargs)
  File "/home/U/UBot/actions.py", line 795, in skip_job
    do_skip_fast(context.bot, player, job_queue)
  File "<string>", line 2, in wrapped
  File "/home/U/UBot/venv/lib/python3.9/site-packages/pony/orm/core.py", line 519, in new_func
    result = func(*args, **kwargs)
  File "/home/U/UBot/internationalization.py", line 126, in wrapped
    user, chat = _user_chat_from_update(update)
  File "/home/U/UBot/internationalization.py", line 155, in _user_chat_from_update
    user = update.effective_user
AttributeError: 'ExtBot' object has no attribute 'effective_user'

By fixing the previous bug, a new one comes up.

**Comment by jh0ker** [comment]
I'll merge this as-is so we can potentially work on some other issues (like supporting the new thread stuff). Thanks a lot for your work, @tehcneko, i really appreciate it even if i'm not very active here anymore :smile:

**Comment by DO97** [comment]
I point out that `actions.py` should still be updated to v13 and `reset_waiting_time()` in `bot.py` as well.

**Pull request merged**